src/dataset/mod.rs

```diff
@@ -323,6 +323,18 @@ pub trait Labels {
     fn labels(&self) -> Vec<Self::Elem> {
         self.label_set().into_iter().flatten().collect()
```

Comment:
For some reason this method doesn't dedup the final vector. It should do something like union all `HashSet` together. Or we can just the return type to `HashSet`, but that might be too invasive.